src/main/webapp/script.js

```diff
@@ -36,152 +44,42 @@ document.addEventListener('DOMContentLoaded', () => {
-  const JPG_LIMIT_MB = 2;
-  if (imageType === 'jpeg' && image.size > JPG_LIMIT_MB * 1024 * 1024) {
-    throw new Error(
-        `File size should not exceed ${JPG_LIMIT_MB}MB for jpeg images.` +
-        'The size of an uploaded jpeg image is ' +
-        Math.ceil(image.size / 1024 / 1024) + 'MB');
-  }
+  return rangeInput;
 }
 
 /**
- * Function to get file type if it is png or jpeg
- * and throw error otherwise
- * @param {File} file
- * @return {Promise}
+ * Function which is called when user chooses file to upload.
+ * @param {Event} event "Change" event on image upload button.
  */
-function getImageTypeOrError(file) {
-  return new Promise(function(resolve, reject) {
-    const fileReader = new FileReader();
-
-    fileReader.onloadend = function(e) {
-      // get first 4 bytes from file - they contain information
-      // about the extension.
-      const bytes = new Uint8Array(e.target.result);
-
-      // convert those bytes to string
-      let header = '';
-      for (let i = 0; i < bytes.length; i++) {
-        header += bytes[i].toString(16);
-      }
-
-      const PNG_HEADERS = ['89504e47'];
-      if (PNG_HEADERS.includes(header)) {
-        resolve('png');
-      }
-
-      const JPEG_HEADERS = ['ffd8ffe0', 'ffd8ffe1',
-        'ffd8ffe2', 'ffd8ffe3', 'ffd8ffe8'];
-      if (JPEG_HEADERS.includes(header)) {
-        resolve('jpeg');
-      }
-
-      reject(new Error(
-          'Invalid file type. Only jpeg and png images can be uploaded'));
-    };
+async function handleImageUpload(event) {
+  await validateImageUpload().then(async () => {
+    // blur the image and update html page.
+    const imageUrl = URL.createObjectURL(event.target.files[0]);
 
-    // read first 4 bytes from file
-    fileReader.readAsArrayBuffer(file.slice(0, 4));
-  });
-}
+    const blurAreas = await getBlurAreas(event.target.files[0]);
 
-/**
- * Function to make sure the image resolution is no
- * more than mxWidth x mxHeight
- * Promise returns error if this is not true and
- * nothing otherwise.
- * @param {File} imageFile
- * @param {Number} mxWidth
- * @param {Number} mxHeight
- * @return {Promise}
- */
-function validateImageResolution(imageFile, mxWidth, mxHeight) {
-  return new Promise(function(resolve, reject) {
-    // construct image object from the file
-    const imageObject = new Image();
-    const imageUrl = URL.createObjectURL(imageFile);
-
-    imageObject.onload = function() {
-      if (imageObject.width * imageObject.height >
-          mxWidth * mxHeight) {
-        reject(new Error(
-            'The image resolution can not exceed ' + mxWidth +
-            'x' + mxHeight + 'px. ' +
-            'The uploaded image resolution is ' +
-            imageObject.width + 'x' + imageObject.height + 'px'));
-        return;
-      }
-
-      resolve();
-    };
-
-    imageObject.src = imageUrl;
+    processImage(imageUrl, blurAreas);
+  }).catch((error) => {
+    alert(error.message);
```

Comment:
Maybe change it to `alert("ERROR: " + error.message);` so the user knows that what happened is not expected and the image was not processed.